components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/subscription/Application.java

```diff
@@ -101,15 +101,25 @@ public Integer getCacheKey() {
         return getId();
     }
 
-    public Map<String, String> getAttributesMap() {
+    public Map<String, String> getAttributes() {
 
-        return attributesMap;
+        return attributes;
     }
 
     public void addAttribute(String name, String value) {
 
-        if (!this.attributesMap.containsKey(name)) {
-            this.attributesMap.put(name, value);
+        if (!this.attributes.containsKey(name)) {
```

Comment:
if an application attribute is updated, we may need to remove and add updatex attribute. To handle that shall we add a remove method as well?

Reply:
no need to done since we remove object and add when updating